rcl/src/rcl/wait.c

```diff
@@ -430,6 +436,17 @@ rcl_wait(rcl_wait_set_t * wait_set, int64_t timeout)
     &dummy_clients,
     wait_set->impl->rmw_waitset,
     timeout_argument);
+  // Check for timeout.
+  if (ret == RMW_RET_TIMEOUT) {
+    // Assume none were set (because timeout was reached first), and clear all.
+    rcl_ret_t rcl_ret;
+    (void)rcl_ret;  // NO LINT
+    rcl_ret = rcl_wait_set_clear_subscriptions(wait_set);
+    assert(rcl_ret == RCL_RET_OK);  // Defensive, shouldn't fail with valid wait_set.
```

Comment:
Do we need this check for release too or should this never ever happen and is only a trip wire?